assets/src/scripts/modules/donate-buttons.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Collect any "Donate" buttons on the page.
+ */
+const donationButtons = [ ...document.querySelectorAll( '[href^="https://donate.wikimedia.org"]' ) ];
+const { object_id } = window.shiro;
+
+/**
+ * Searches for any links to donate.wikimedia.org, and replaces their hrefs to
+ * include UTM parameters in the query strings.
+ *
+ * @returns {HTMLElement[]} Array of donation buttons.
+ */
+const init = () => donationButtons.forEach(
```

Comment:
Because we're spreading the qsa results into an array, it will always be at worst `[]` -- which does have a forEach method, so it would execute safely even if there are no buttons. You can test this yourself with

```js
[ ...document.querySelectorAll( '.does-not-exist' ) ]
    .forEach( () => console.log( 'Should not run' ) );
```
which should execute without error.

In most modern browsers, the qsa result itself (which is a [NodeList](https://developer.mozilla.org/en-US/docs/Web/API/NodeList)) has a `forEach` method too—so this should also work safely:

```js
document.querySelectorAll( '.does-not-exist' )
    .forEach( () => console.log( 'Should not run' ) );
```

The array spread is a safeguard for compatibility which we _may_ not need, but it's still useful in my opinion. It's easier to think about "what can arrays do" than "what can a NodeList do".